src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/FormFileModelBinder.cs

```diff
@@ -43,15 +43,22 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
 
         private async Task<ModelBindingResult> BindModelCoreAsync(ModelBindingContext bindingContext)
         {
+            // If we're at the top level, then use the FieldName (paramter or property name).
+            // This handles the fact that there will be nothing in the ValueProviders for this parameter
+            // and so we'll do the right thing even though we 'fell-back' to the empty prefix.
+            var modelName = bindingContext.IsTopLevelObject
+                ? bindingContext.FieldName
```

Comment:
Thinking of what you did in `BodyModelBinder`, should this use `BinderModelName` if set?